src/ERC6538Registry.sol

```diff
@@ -24,22 +58,185 @@ contract ERC6538Registry is IERC6538Registry {
     uint256 schemeId,
     bytes memory signature,
     bytes memory stealthMetaAddress
-  ) external pure {
-    registerKeysOnBehalf(_toBytes(registrant), schemeId, signature, stealthMetaAddress);
+  ) external {
+    // Check for nonce
+    bytes32 digest = _hashTypedDataV4(
+      keccak256(
+        abi.encode(TYPE_HASH, registrant, schemeId, stealthMetaAddress, nonceOf[registrant]++)
+      )
+    );
+    require(isValidSignatureNow(registrant, digest, signature), "Invalid signature");
+    stealthMetaAddressOf[registrant][schemeId] = stealthMetaAddress;
+    emit StealthMetaAddressSet(registrant, schemeId, stealthMetaAddress);
   }
 
-  /// @inheritdoc IERC6538Registry
-  function registerKeysOnBehalf(
-    bytes memory, // registrant
-    uint256, // schemeId
-    bytes memory, // signature
-    bytes memory // stealthMetaAddress
-  ) public pure {
-    revert("not implemented");
+  /**
+   * @dev Returns the domain separator for the current chain.
+   */
+  function _domainSeparatorV4() public view returns (bytes32) {
+    if (address(this) == _cachedThis && block.chainid == _cachedChainId) {
+      return _cachedDomainSeparator;
+    } else {
+      return _buildDomainSeparator();
+    }
+  }
+
+  function _buildDomainSeparator() private view returns (bytes32) {
+    return
+      keccak256(abi.encode(TYPE_HASH, _hashedName, _hashedVersion, block.chainid, address(this)));
+  }
+
+  /**
+   * @dev Given an already https://eips.ethereum.org/EIPS/eip-712#definition-of-hashstruct[hashed
+   * struct], this
+   * function returns the hash of the fully encoded EIP712 message for this domain.
+   *
+   * This hash can be used together with {ECDSA-recover} to obtain the signer of a message. For
```

Comment:
Nit, this is OZ's custom natspec format for the docgen parser, so we should edit this natspec

Reply:
OZ's natspec style and unexpected line breaks still appear in a lot of places

Reply:
My bad, went through line by line and updated the documentation here: https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5/commits/12039aa69c458db0a3b464256fe94d3eded88443 If there are still unexpected line breaks it might be my formatter settings.